db/db_impl_write.cc

```diff
@@ -534,23 +566,76 @@ Status DBImpl::PipelinedWriteImpl(const WriteOptions& write_options,
   return w.FinalStatus();
 }
 
+Status DBImpl::UnorderedWriteMemtable(const WriteOptions& write_options,
+                                      WriteBatch* my_batch,
+                                      WriteCallback* callback, uint64_t log_ref,
+                                      SequenceNumber seq) {
+  PERF_TIMER_GUARD(write_pre_and_post_process_time);
+  StopWatch write_sw(env_, immutable_db_options_.statistics.get(), DB_WRITE);
+
+  WriteThread::Writer w(write_options, my_batch, callback, log_ref,
+                        false /*disable_memtable*/);
```

Comment:
I'm a little bit confused. Do we create a Writer object but never enter WriteThread? If my understanding is correct, what's the point of creating such an object?

Reply:
All the ::InsertInto functions in this file are receiving a writer object (so does ::CheckCallBack). I found it simpler to write and read the code if we stick to that here. But I do not have a strong opinion here.